primitives/src/v2/mod.rs

```diff
@@ -1569,6 +1573,114 @@ impl CompactStatement {
 	}
 }
 
+/// `Validators` struct indexed by `ValidatorIndex`.
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo, Default)]
+#[cfg_attr(feature = "std", derive(PartialEq, MallocSizeOf))]
+pub struct Validators(Vec<ValidatorId>);
+
+impl From<Vec<ValidatorId>> for Validators {
+	fn from(validators: Vec<ValidatorId>) -> Self {
+		Validators(validators)
+	}
+}
+
+impl FromIterator<ValidatorId> for Validators {
+	fn from_iter<T: IntoIterator<Item = ValidatorId>>(iter: T) -> Self {
+		Self(Vec::from_iter(iter))
+	}
+}
+
+impl Validators {
```

Comment:
We also have other keys indexed by `ValidatorIndex`, something more generic might make sense here with `Validators` being a type alias for something like `IndexedVec<ValidatorIndex, ValidatorId>`.

Other than that looks really good at first sight.

Reply:
Actually considering the generic version. Will look to implementing it.